compiler/rustc_trait_selection/src/traits/error_reporting/on_unimplemented.rs

```diff
@@ -111,6 +112,8 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
         trait_ref: ty::PolyTraitRef<'tcx>,
         obligation: &PredicateObligation<'tcx>,
     ) -> OnUnimplementedNote {
+        let mut long_ty_file = None;
```

Comment:
Where is the path of this file being printed?

Reply:
So this is a bit awkward, I tried to make sure I don't miss any silent type name being written to file, but now there seems to be a duplicate written to file note lol

Reply:
That... should be fine for now. We need to bubble the printing of the file (and passing the file down) more than we do now, but it being duplicated is better than out right losing it.

Reply:
> That... should be fine for now. We need to bubble the printing of the file (and passing the file down) more than we do now, but it being duplicated is better than out right losing it.

Yeah, there's `let mut file = None` a bit all over the place right now lol